nativelib/src/main/resources/scala-native/delimcc.c

```diff
@@ -0,0 +1,377 @@
+    void **new_return_slot; // new return slot
+    lh_jmp_buf return_buf;
+
+    target = tail - continuation->size;
+    diff = target - continuation->stack_top;
+    // set up stuff
+    memcpy(return_buf, continuation->buf, ASM_JMPBUF_SIZE);
+
+#if !defined(__i386__) // 32 bit platforms don't have an alignment restriction?
+    assert((diff & 15) == 0);
+#endif
+    // fprintf(stderr,
+    //         "diff is %td, stack (size = %td) goes %p~%p -> %p~%p | original "
+    //         "cont = %p [%p]\n",
+    //         diff, cont->size, cont->stack_top, cont->stack_top + cont->size,
+    //         target, tail, cont, cont->stack);
+    // clone the handler chain, with fixes.
+    to_install = nw = handler_clone_fix(continuation->handlers, diff);
+    jmpbuf_fix(return_buf);
+    // copy and fix the remaining information in the continuation
+    new_return_slot = fixed_addr(continuation->return_slot);
+    // install the memory
+    memcpy(target, continuation->stack, continuation->size);
+    // fix the handlers in cont->stack
+    for (i = 0; i < continuation->handlers_len; ++i, nw = nw->next) {
+        fix_addr(nw->h->result);
+        if (nw->h->stack_btm != NULL)
+            fix_addr(nw->h->stack_btm);
+        jmpbuf_fix(nw->h->buf);
+    }
+    // install the handlers and fix the return buf
+    handler_install(to_install);
+
+    // set return value for the return slot
+    // fprintf(stderr, "return slot is %p\n", new_return_slot);
+    *new_return_slot = out;
+    // fix the return address of the bottom of our new stack fragment.
+    *(void **)(target + continuation->size - BOUNDARY_LR_OFFSET) = ret_addr;
+    _lh_longjmp(return_buf, 1);
+}
+
+void *scalanative_continuation_resume(Continuation *continuation, void *out) {
+    /*
+     * Why we need a setjmp/longjmp.
+     *
+     * `resume` stack doesn't know which registers are changed, and so we
+     * basically need to save all of them. setjmp/longjmp-ing to the same place
+     * is the easiest way to do so.
+     *
+     * Why we need a Handler.
+     *
+     * Resumed computation might suspend on a parent, and mess up the setjmp
+     * buffer that way.
+     * */
+    volatile void *result = NULL; // we need to force the compiler to re-read
+    // this from stack every time.
+    volatile ContinuationBoundaryLabel label = next_label_count();
+    Handler h = {.id = label, .result = &result, .stack_btm = NULL};
+    handler_push(&h);
+    if (_lh_setjmp(h.buf) == 0) {
+        result = _lh_resume_entry(continuation->size, continuation, out);
+        handler_head_longjmp(1); // top handler is always ours, avoid
+                                 // refering to non-volatile `h`
+    }
+    handler_pop(label);
+    return (void *)result;
+}
+
+static void handler_free(Handlers *hs) {
+    while (hs != NULL) {
+        Handlers *old = hs;
+        hs = hs->next;
+        free(old);
+    }
+}
+
+void scalanative_continuation_free(Continuation *continuation) {
```

Comment:
Seems like it's never used

Reply:
Yep, this is only used when a custom allocation function (in Scala Native case, with ObjectArrays) is not used. 
It was useful for writing C code that uses `delimcc.c` directly, maybe I should guard that under some `#ifdef`.